### ISSUE TITLE
[iOS] Incompatible with latest Firebase package (iOS SDK v9+/RN package v15+)

### ISSUE THREAD

**Issue by lostchopstik** [Tealium/tealium-react-native#129]
**Environment**:
MacOS 12.4
XCode 13.4.1
Macbook Pro M1 2020

**Plugin Version**: 2.2.0 and 2.2.1

**Issue**:
I cannot get a successful build on my M1 (arm64) Mac. I get the following error:
```
Undefined symbols for architecture arm64:

  "_OBJC_CLASS_$_TealiumReactNative", referenced from:

      objc-class-ref in TealiumReactNative.o

      __OBJC_$_CATEGORY_TealiumReactNative_$_RCTExternModule in TealiumReactNative.o

  "_OBJC_CLASS_$_TealiumWrapper", referenced from:

      objc-class-ref in TealiumReactNative.o

      __OBJC_$_CATEGORY_TealiumWrapper_$_RCTExternModule in TealiumReactNative.o

ld: symbol(s) not found for architecture arm64

clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Possible Cause**:
I originally had upgraded from 2.2.0 to 2.2.1 and simultaneously upgraded my React Native Firebase packages from 12.2.0 to 15.2.0. Starting with the RNFB 15+, they require adding `use_frameworks!` to the Podfile. After reverting back to 12.2.0 and removing the `use_frameworks!` line, I was able to build again. No other packages reported similar errors.

[https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md#breaking-changes](https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md#breaking-changes)

**Comment by craigrouse** [comment]
@lostchopstik we're looking into this. We'll get back to you as soon as we can.

**Comment by jameskeith** [comment]
hi @lostchopstik 
Apologies that it's taken a while to get back to you on this. I have been able to reproduce the issue described after importing the latest `tealium-react-native` package as well as the latest `@react-native-firebase/app` package and enabling `use_frameworks!` 

I have been able to get it building and running locally by forcing it to link statically either via :
```
use_frameworks! :linkage => :static
```
or by linking tealium-react-native package statically only:
```
pre_install do |installer|
    Pod::Installer::Xcode::TargetValidator.send(:define_method, :verify_no_static_framework_transitive_dependencies) {}

    installer.pod_targets.each do |pod|
      if pod.name.start_with?('tealium-react-native')
        def pod.build_type;
          Pod::BuildType.static_library # >= 1.9
        end
      end
    end
  end
```

hopefully that should get your project building - I believe that our module is expected to support dynamic linking, so we'll need to take a look at that in a future release.

Hope that's all useful, let me know how you get on.
